### DISCUSSION TITLE
Unable to create down.sql for first migration

### DISCUSSION THREAD

**Discussion by hariria** [prisma/prisma#21011] [Q&A] [answered]
### Question

Following the guide here: https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/generating-down-migrations for generating down migrations.

### Context

The article recommends that you do the following to create `down.sql` migrations.

1. Make a change to your `prisma.schema`
2. Run `migrate diff` command

```bash
pnpx prisma migrate diff \
 --from-schema-datamodel prisma/schema.prisma \
 --to-schema-datasource prisma/schema.prisma \
 --script > down.sql
```

3. A `down.sql` will be generated at the root

### Problem

The problem is that this works well if you have already run a migration once, but it returns `-- This is an empty migration.` for `down.sql` on the first migration.

How am I supposed to generate a `down.sql` file for the first migration?

### Expected behavior (optional)

I expect `migrate diff` to be able to generate a `down.sql` file on the first migration with things like `DROP __SOME_TABLE__`.

### Information about Prisma Schema, Client Queries and Environment (optional)

Starting `prisma.schema` looks like this:

```prisma
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id      Int     @id @default(autoincrement())
  email   String  @unique
  name    String?
  another String
}
```

**Comment by nikolasburk** [comment]
Hey @hariria 👋 

Thanks a lot for raising this question!

I have been unable to reproduce the problem though. Here's what I did:

1. Use `npx try-prisma@latest --template typescript/script -i npm -p . -n test-down-migrations` to scaffold a new project.
2. Change the `datasource` to `postgresql` and update the `url` to point to a local PostgreSQL server: `postgresql://nikolasburk:nikolasburk@localhost:5432/down-migrations`
3. Manually create a DB called `down-migrations` on that PostgreSQL server.
4. Update the `schema.prisma` file to look identical to the one from your question.
5. Run the command from your question (though using `npx` and not `pnpx`, but I don't think that should matter?!):
    ```
    npx prisma migrate diff \
      --from-schema-datamodel prisma/schema.prisma \
      --to-schema-datasource prisma/schema.prisma \
      --script > down.sql
    ```
6. The result was a file called `down.sql` with the following contents:
    ```sql
    -- DropTable
    DROP TABLE "User";
    ```

Are you able to follow these steps and get to the same result? Could you maybe share a small reproduction of your issue in a dedicated repo so that we can look into it further?

Please let me know if you have any further questions, always happy to help you out!

**Comment by hariria** [comment, marked as answer]
@nikolasburk thanks for the response, i figured out the issue. The `postgres` db i made already contained the tables from before (was not a new db). I thought prisma generated the `down.sql` on the diff between the previous schema and the new schema. Anyways thanks for letting me know, would be great if it was made a bit more clear in the docs, I see it's stated [in the generating down migrations guide](https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/generating-down-migrations) as 

> Using --to-schema-datasource: this makes a comparison to the state of the database.

Maybe this could be a bit clearer or some other warning could be shown if the schema does not match the db?
